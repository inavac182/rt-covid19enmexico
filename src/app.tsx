import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { createBrowserHistory } from 'history';
import { BrowserRouter, Route, Switch } from 'react-router-dom';

import { RouterStore, syncHistoryWithStore } from 'mobx-react-router';
import { Provider } from 'mobx-react';

import { CasesStore } from './stores';
import { HomePage } from './components';
import './less/main.less';
import { Router } from 'react-router';

const browserHistory = createBrowserHistory();
const routingStore = new RouterStore();
const stores = {
  routing: routingStore,
  casesStore: new CasesStore(),
};

const history = syncHistoryWithStore(browserHistory, routingStore);

export default function MainRouter() {
  return (
    <BrowserRouter>
      <Provider {...stores}>
        <Router history={history}>
          <Switch>
            <Route exact path="/" component={HomePage} />
          </Switch>
        </Router>
      </Provider>
    </BrowserRouter>
  );
}

ReactDOM.render(<MainRouter />, document.getElementById('main'));
