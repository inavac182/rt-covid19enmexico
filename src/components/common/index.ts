export * from './checkbox-button';
export * from './alert';
