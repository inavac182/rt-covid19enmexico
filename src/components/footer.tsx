import React, { useEffect } from 'react';
import { inject, observer } from 'mobx-react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTwitter } from '@fortawesome/free-brands-svg-icons';

interface FooterProps {
  hideLastUpdate?: boolean;
}

export const Footer = (props: FooterProps) => {
  const { hideLastUpdate } = props;

  return (
    <section className="footer">
      <h3>Fuentes:</h3>
      <p>
        Secretaria de salud:{' '}
        <a href="https://www.gob.mx/salud/es/archivo/documentos" target="_blank">
          Ir al sitio
        </a>
      </p>
      <p className="disclaimer">
        Estos datos se actualizan con respecto a los comunicados emitidos por los gobiernos estatales y confirmados
        diariamente con el informe de la Secretaria de Salud Federal.
      </p>
      <div className="contant-data">
        <p>Supporters:</p>
        <p>
          Felipe:{' '}
          <a href="https://twitter.com/FelipeNava92" target="_blank">
            <FontAwesomeIcon icon={faTwitter} /> @FelipeNava92
          </a>
        </p>
        <p>
          Jaqueline:{' '}
          <a href="https://twitter.com/Jackelyn2808" target="_blank">
            <FontAwesomeIcon icon={faTwitter} /> @Jackelyn2808
          </a>
        </p>
      </div>
    </section>
  );
};
