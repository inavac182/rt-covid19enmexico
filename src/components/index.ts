export * from './dashboard';
export * from './home-page';
export * from './header';
export * from './footer';
export * from './info-modal';
