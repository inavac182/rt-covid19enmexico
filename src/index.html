<!DOCTYPE html>
<html>
    <head profile="https://firebasestorage.googleapis.com/v0/b/mapa-covid19-mexico.appspot.com/o/mexico.jpg?alt=media&token=3a7ecc1f-593c-4672-9b84-de9711517d41">
        <meta charset="UTF-8" />
        <title>RT COVID-19</title>
        <link rel="icon" type="image/jpg" href="https://firebasestorage.googleapis.com/v0/b/mapa-covid19-mexico.appspot.com/o/mexico.jpg?alt=media&token=3a7ecc1f-593c-4672-9b84-de9711517d41">
        <meta name="viewport" content="width=device-width,initial-scale=1">
        <meta name="Description" CONTENT="Autor: Felipe Nava, Mapa Ilustrativo de los casos confirmados de COVID-19 Corona virus en México.">
        <meta name="google" content="notranslate" />
        <meta property="og:title" content="Estadísticas del covid-19 (Corona virus) en México">
        <meta property="og:description" content="Sitio web con actualización en tiempo real de casos confirmados de corona virus en México.">
        <meta property="og:url" content="http://covid19enmexico.com/">
        <meta property="og:image" content="https://firebasestorage.googleapis.com/v0/b/mapa-covid19-mexico.appspot.com/o/thumb7.png?alt=media&token=2c929232-20d1-434a-b46d-1d8343432f51">
        <meta name="twitter:title" content="Estadísticas del covid-19 (Corona virus) en México">
        <meta name="twitter:description" content="Sitio web con actualización en tiempo real de casos confirmados de corona virus en México.">
        <meta name="twitter:card" content="summary_large_image">
        <meta name="twitter:image" content="https://firebasestorage.googleapis.com/v0/b/mapa-covid19-mexico.appspot.com/o/thumb7.png?alt=media&token=2c929232-20d1-434a-b46d-1d8343432f51">
        <link href="https://fonts.googleapis.com/css?family=Bellota|Anton|Sen&display=swap" rel="stylesheet">
    </head>

    <body>
        <div id="main"></div>
    </body>
</html>
