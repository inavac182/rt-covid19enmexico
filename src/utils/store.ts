import { toJS, runInAction } from 'mobx';
import { StoreDefaults } from './storeDefaults';
import { withoutKeys } from './merge';

export abstract class Store {
  public static DEFAULTS?: StoreDefaults;

  public constructor(state: StoreDefaults = {}) {
    const defaults = this.constructor['DEFAULTS'] || {};

    runInAction(() => Object.assign(this, defaults, state));
  }

  public abstract hydrate(data: StoreDefaults): void;

  public toJSON(ignoreAdditionalKeys: string[] | string = []): StoreDefaults {
    if (!Array.isArray(ignoreAdditionalKeys)) {
      ignoreAdditionalKeys = ignoreAdditionalKeys.constructor === String ? [ignoreAdditionalKeys] : [];
    }

    const regexIgnorePrivate = /^_.+/;
    const simpleObject: any = toJS(this);
    const ignoreKeys: string[] = Object.keys(simpleObject)
      .filter(key => regexIgnorePrivate.test(key) || typeof simpleObject[key] === 'function')
      .concat(ignoreAdditionalKeys, 'logger');

    return withoutKeys(ignoreKeys).merge({}, simpleObject);
  }
}
