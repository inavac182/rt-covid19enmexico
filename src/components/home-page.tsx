import React, { useEffect, useState } from 'react';
import { observer, inject } from 'mobx-react';
import { RouterStore } from 'mobx-react-router';

import { MatchProps } from 'src/types';
import { CasesStore } from 'src/stores';
import { Dashboard } from '.';

interface HomePageProps {
  casesStore?: CasesStore;
  routing?: RouterStore;
  match: MatchProps;
}

export const HomePage = inject(
  'casesStore',
  'routing'
)(
  observer((props: HomePageProps) => {
    const { casesStore } = props;
    const [loading, setLoading] = useState(true);

    useEffect(() => {
      casesStore.fetch(() => {
        setLoading(false);
      });
    }, []);

    return <>{loading ? <p>Loading...</p> : <Dashboard />}</>;
  })
);
