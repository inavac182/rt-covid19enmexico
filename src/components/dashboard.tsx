import React, { useState, useEffect } from 'react';
import { inject } from 'mobx-react';
import { Header, Footer } from '.';
import { ModalContext } from './modal';
import { Alert } from './common';
import { faInfoCircle } from '@fortawesome/free-solid-svg-icons';

interface DashboardProps {}

export const Dashboard = (props: DashboardProps) => {
  const [isModalOpen, setModalOpen] = useState(false);
  const contextHandler = {
    isModalOpen,
    toggleModal: () => {
      setModalOpen(!isModalOpen);
    },
  };

  useEffect(() => {
    document.title = `RT del COVID-19 en México`;
  }, []);

  return (
    <ModalContext.Provider value={contextHandler}>
      <section className={`dashboard modal-${isModalOpen}`}>
        <Header />
        <section className="main">
          <Alert
            icon={faInfoCircle}
            text="Esta sección está en progreso."
            classes="warning small center margin-top-three"
          />
        </section>
        <Footer />
      </section>
    </ModalContext.Provider>
  );
};
