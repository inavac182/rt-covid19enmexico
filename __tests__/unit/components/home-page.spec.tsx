import * as React from 'react';
import { HomePage } from '../../../src/components/';
import { HomePageStore } from '../../../src/stores/home-page.store';
import { TestProvider } from '../../../src/utils/testProvider';

describe ('HomePage', () => {
    const stores = {
      homePageStore: new HomePageStore()
    };

    it('renders correctly', () => {
      const homepage = TestProvider(stores).render(<HomePage match={{}} />);
      expect(homepage).toMatchSnapshot();
    });

    it('should render home page with correct title', () => {
      const expectedTitle = 'My Title';
      const stores = {
        homePageStore: new HomePageStore({
          title: expectedTitle
        })
      };

      const homePageWrapper = TestProvider(stores).mount(<HomePage  match={{}} />);
      const title = homePageWrapper.find('h1');

      expect(title.text()).toBe(expectedTitle);
    });
});

