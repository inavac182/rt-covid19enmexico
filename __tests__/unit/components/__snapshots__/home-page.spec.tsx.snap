// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage renders correctly 1`] = `
<div
  class="page"
>
  <h1>
    Cova trader
  </h1>
  <div
    class="books"
  >
    <p>
      Loading books....
    </p>
  </div>
</div>
`;
